samza-core/src/main/scala/org/apache/samza/config/StreamConfig.java

```diff
@@ -0,0 +1,394 @@
+  }
+
+  private List<String> getStreamIdsForSystem(String system) {
+    return getStreamIds().stream().filter(streamId -> system.equals(getSystem(streamId))).collect(Collectors.toList());
+  }
+
+  /**
+   * Finds the stream id which corresponds to the systemStream.
+   * This finds the stream id that is mapped to the system in systemStream through the config and that has a physical
+   * name (the physical name might be the stream id itself if there is no explicit mapping) that matches the stream in
+   * systemStream.
+   * Note: If the stream in the systemStream is a stream id which is mapped to a physical stream, then that stream won't
+   * be returned as a stream id here, since the stream in systemStream doesn't match the physical stream name.
+   *
+   * @param systemStream system stream to map to stream id
+   * @return             stream id corresponding to the system stream
+   */
+  public String systemStreamToStreamId(SystemStream systemStream) {
+    List<String> streamIds = getStreamIdsForSystem(systemStream.getSystem()).stream()
+      .filter(streamId -> systemStream.getStream().equals(getPhysicalName(streamId))).collect(Collectors.toList());
+    if (streamIds.size() > 1) {
+      throw new IllegalStateException(String.format("There was more than one stream found for system stream %s", systemStream));
+    }
+
+    return streamIds.isEmpty() ? null : streamIds.get(0);
+  }
+
+  /**
+   * Gets the specified Samza property for a SystemStream. A Samza property is a property that controls how Samza
+   * interacts with the stream, as opposed to a property of the stream itself.
+   *
+   * First, tries to map the systemStream to a streamId. This will only find a streamId if the stream is a physical name
+   * (explicitly mapped physical name or a stream id without a physical name mapping). That means this will not map a
+   * stream id to itself if there is a mapping from the stream id to a physical stream name. This also requires that the
+   * stream id is mapped to a system in the config.
+   * If a stream id is found:
+   * 1) Look for "streams.{streamId}.{property}" for the stream id.
+   * 2) Otherwise, look for "systems.{systemName}.streams.{streamName}.{property}" in which the systemName is the system
+   * mapped to the stream id and the streamName is the physical stream name for the stream id.
+   * 3) Otherwise, look for "systems.{systemName}.default.stream.{property}" in which the systemName is the system
+   * mapped to the stream id.
+   * If a stream id was not found or no property could be found using the above keys:
+   * 1) Look for "systems.{systemName}.streams.{streamName}.{property}" in which the systemName is the system in the
+   * input systemStream and the streamName is the stream from the input systemStream.
+   * 2) Otherwise, look for "systems.{systemName}.default.stream.{property}" in which the systemName is the system
+   * in the input systemStream.
+   *
```

Comment:
I forgot to add that this can return `null`. Could you please add that?

Reply:
you mean in the comment? As 3) return null.

Reply:
Yep